src/otx/core/data/dataset/object_detection_3d.py

```diff
@@ -0,0 +1,307 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTX3DObjectDetectionDataset."""
+
+# mypy: ignore-errors
+
+from __future__ import annotations
+
+from copy import deepcopy
+from functools import partial
+from typing import TYPE_CHECKING, Any, Callable, List, Union
+
+import numpy as np
+import torch
+from datumaro import Image
+from PIL import Image as PILImage
+from torchvision import tv_tensors
+
+from otx.core.data.dataset.utils.kitti_utils import Calibration, affine_transform, angle2class, get_affine_transform
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.object_detection_3d import Det3DBatchDataEntity, Det3DDataEntity
+from otx.core.data.mem_cache import NULL_MEM_CACHE_HANDLER, MemCacheHandlerBase
+from otx.core.data.transform_libs.torchvision import Compose
+from otx.core.types.image import ImageColorChannel
+
+from .base import OTXDataset
+
+if TYPE_CHECKING:
+    from datumaro import Bbox, DatasetSubset
+
+
+Transforms = Union[Compose, Callable, List[Callable], dict[str, Compose | Callable | List[Callable]]]
+
+
+class OTX3DObjectDetectionDataset(OTXDataset[Det3DDataEntity]):
+    """OTXDataset class for detection task."""
+
+    def __init__(
+        self,
+        dm_subset: DatasetSubset,
+        transforms: Transforms,
+        mem_cache_handler: MemCacheHandlerBase = NULL_MEM_CACHE_HANDLER,
+        mem_cache_img_max_size: tuple[int, int] | None = None,
+        max_refetch: int = 1000,
+        image_color_channel: ImageColorChannel = ImageColorChannel.RGB,
+        stack_images: bool = True,
+        to_tv_image: bool = True,
+        max_objects: int = 50,
+        depth_threshold: int = 65,
+        resolution: tuple[int, int] = (1280, 384),  # (W, H)
```

Comment:
It seems that it should be same as input_size of data module, right? If them could you update `OTXDatasetFactory` to pass input_size to this class? I think input_size should be aligned automatically when user change the input_size data.input_size in recipe.

Reply:
It will requires a bit more efforts, since 3D task needs augmentation also for 3d meta data including angles. If we want to use our OTX data pipeline we must align all of the annotations. I plan to do that PR soon.